src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -60,109 +141,85 @@ export class CalciteSwitch {
     }
   }
 
-  @Listen("click") onClick(e: MouseEvent): void {
+  @Listen("click")
+  onClick(e: MouseEvent): void {
     // prevent duplicate click events that occur
     // when the component is wrapped in a label and checkbox is clicked
     if (
-      (!this.disabled && this.el.closest("label") && e.target === this.inputProxy) ||
+      (!this.disabled && this.el.closest("label") && e.target === this.inputEl) ||
       (!this.el.closest("label") && e.target === this.el)
     ) {
-      this.updateSwitch(e);
+      this.toggle();
     }
   }
 
-  @Listen("keydown") keyDownHandler(e: KeyboardEvent): void {
+  @Listen("keydown")
+  keyDownHandler(e: KeyboardEvent): void {
     const key = getKey(e.key);
     if (!this.disabled && (key === " " || key === "Enter")) {
-      this.updateSwitch(e);
+      this.toggle();
     }
   }
 
-  @Watch("switched") switchWatcher(): void {
-    this.switched
-      ? this.inputProxy.setAttribute("checked", "")
-      : this.inputProxy.removeAttribute("checked");
-  }
-
-  private inputProxy: HTMLInputElement;
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
 
   connectedCallback(): void {
-    this.setupProxyInput();
+    this.guid = this.el.id || `calcite-switch-${guid()}`;
```

Comment:
I'm not sure if this is a big deal, but the guid will change if a component is reconnected.  Maybe it should be in `componentWillLoad`? 

https://stenciljs.com/docs/component-lifecycle